DevEdu.API/DevEdu.Business.Tests/LessonServiceTests.cs

```diff
@@ -0,0 +1,195 @@
+
+            //When
+            var actual = sut.SelectAllLessonsByGroupId(groupId);
+
+            //Then
+            Assert.AreEqual(expected, actual);
+            _lessonRepoMock.Verify(x => x.SelectAllLessonsByGroupId(groupId), Times.Once);
+        }
+
+        [Test]
+        public void SelectAllLessons_IntTeacherId_ListOfLessonDto()
+        {
+            //Given
+            var expected = LessonData.GetLessons();
+
+            var teacherId = 3;
+
+            _lessonRepoMock.Setup(x => x.SelectAllLessonsByTeacherId(teacherId)).Returns(expected);
+
+            var sut = new LessonService(_lessonRepoMock.Object, _commentRepoMock.Object, _userRepoMock.Object,
+                _userValidationHelperMock.Object, _lessonValidationHelperMock.Object);
+
+            //When
+            var actual = sut.SelectAllLessonsByTeacherId(teacherId);
+
+            //Then
+            Assert.AreEqual(expected, actual);
+            _lessonRepoMock.Verify(x => x.SelectAllLessonsByTeacherId(teacherId), Times.Once);
+        }
+
+        [Test]
+        public void SelectLesson_IntLessonId_LessonDto()
+        {
+            //Given
+            var expected = LessonData.GetSelectedLessonDto();
+
+            var lessonId = LessonData.LessonId;
+
+            _lessonRepoMock.Setup(x => x.SelectLessonById(lessonId)).Returns(expected);
+
+            var sut = new LessonService(_lessonRepoMock.Object, _commentRepoMock.Object, _userRepoMock.Object,
+                _userValidationHelperMock.Object, _lessonValidationHelperMock.Object); 
+
+            //When
+            var actual = sut.SelectLessonById(lessonId);
+
+            //Then
+            Assert.AreEqual(expected, actual);
+            _lessonRepoMock.Verify(x => x.SelectLessonById(lessonId), Times.Once);
+        }
+
+        [Test]
+        public void SelectLessonWithComments_IntLessonId_LessonDto()
+        {
+            //Given
+            var lesson = LessonData.GetSelectedLessonDto();
+            var comments = CommentData.GetListCommentsDto();
+
+            var expected = lesson;
+            expected.Comments = comments;
+
+            var lessonId = LessonData.LessonId;
+
+            _lessonRepoMock.Setup(x => x.SelectLessonById(lessonId)).Returns(lesson);
+            _commentRepoMock.Setup(x => x.SelectCommentsFromLessonByLessonId(lessonId)).Returns(comments);
+
+            var sut = new LessonService(_lessonRepoMock.Object, _commentRepoMock.Object, _userRepoMock.Object,
+                _userValidationHelperMock.Object, _lessonValidationHelperMock.Object);
+
+            //When
+            var actual = sut.SelectLessonWithCommentsById(lessonId);
+
+            //Then
+            Assert.AreEqual(expected, actual);
+            _lessonRepoMock.Verify(x => x.SelectLessonById(lessonId), Times.Once);
+            _commentRepoMock.Verify(x => x.SelectCommentsFromLessonByLessonId(lessonId), Times.Once);
```

Comment:
in addition, please check that SelectStudentsLessonByLessonId is not being called in this case